### PR TITLE
pythonPackages.notmuch2: init at 0.31

### PR DESCRIPTION
**Pull request by teto**
generates bindings via cffi

###### Motivation for this change
wanted to test alot master. I don't think these bindings have been released yet, hence the draft status.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -4114,6 +4114,7 @@ in {
   notify-py = callPackage ../development/python-modules/notify-py { };
 
   notmuch = callPackage ../development/python-modules/notmuch { inherit (pkgs) notmuch; };
+  notmuch2 = callPackage ../development/python-modules/notmuch/2.nix { inherit (pkgs) notmuch; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:4117]:
```suggestion
  notmuch = callPackage ../development/python-modules/notmuch { inherit (pkgs) notmuch; };

  notmuch2 = callPackage ../development/python-modules/notmuch/2.nix { inherit (pkgs) notmuch; };
```

**Comment by lucc** [comment]
In what sense are these bindings "not yet released"? They are mentioned in the release notes for 0.30: https://notmuchmail.org/news/release-0.30/

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (teto) [reply pkgs/top-level/python-packages.nix:4117]: closely related packages tend to be clumped up together, look at neovim/xen/linux/win-qemu in all-packages.nix so I think it's fine.

**Comment by teto** [comment]
@lucc I was just trying to fix my alot and had not checked that part thanks  So it's ready for review then :)

**Comment by teto** [comment]
I have been using it with a lot for a week with no issue.

**Pull request merged**